vassal-app/src/main/java/VASSAL/configure/ConfigureTree.java

```diff
@@ -1345,86 +1517,151 @@ public SearchAction(ConfigureTree configureTree, SearchParameters searchParamete
 
     @Override
     public void actionPerformed(ActionEvent e) {
-      final JDialog d = new JDialog((Frame) SwingUtilities.getAncestorOfClass(Frame.class, configureTree), true);
-      d.setTitle(configureTree.getSearchCmd());
 
-      final JLabel searchLabel = new JLabel("String to find: ");
-      final JTextField search = new JTextField(searchParameters.getSearchString(), 32);
-      search.select(0, searchParameters.getSearchString().length()); // Pre-select all the search text when opening the dialog
-      searchLabel.setLabelFor(search);
-
-      final JCheckBox sensitive = new JCheckBox(Resources.getString("Editor.search_case"), searchParameters.isMatchCase());
-      final JCheckBox names = new JCheckBox(Resources.getString("Editor.search_names"), searchParameters.isMatchNames());
-      final JCheckBox types = new JCheckBox(Resources.getString("Editor.search_types"), searchParameters.isMatchTypes());
-
-      final JButton find = new JButton(Resources.getString("Editor.search_next"));
-      find.addActionListener(new ActionListener() {
-        @Override
-        public void actionPerformed(ActionEvent e) {
+      JDialog d = configureTree.getSearchDialog();
+      JTextField search;
+      if (d != null) {
+        search = configureTree.getSearchField();
+      }
+      else {
+        d = new JDialog((Frame) SwingUtilities.getAncestorOfClass(Frame.class, configureTree), false);
+        configureTree.setSearchDialog(d);
+
+        d.setTitle(configureTree.getSearchCmd());
+
+        final JLabel searchLabel = new JLabel("String to find: ");
```

Comment:
i18n?

Reply:
For sure, but we have a top-to-bottom i18n pass going for these things in separate PRs.